internal/output/markdown.go

```diff
@@ -107,6 +108,13 @@ func ToMarkdown(out Root, opts Options, markdownOpts MarkdownOptions) ([]byte, e
 		return nil, errors.Wrap(err, "Failed to generate diff")
 	}
 
+	diffMsg := ui.StripColor(string(diff))
+	if opts.truncateLength > 0 {
+		truncateMsg := "\n\n...(truncated due to message size limit)...\n\n"
+		truncateLength := utf8.RuneCountInString(diffMsg) - opts.truncateLength - 1000
+		diffMsg = truncateMiddle(diffMsg, truncateLength, truncateMsg)
+	}
```

Comment:
So I know I suggested this `truncateLength` approach, but here's an alternative. You could to truncate the msg before the second `ToMarkdown` call and pass it as an optional `opts.diffMsg`. If this is set, you could skip the `ToDiff` call at the top of the function and use this. Doing this removes the unnecessary `ToDiff` call on the second `ToMarkdown` call.

I'll leave it to you to decide what to do tho.

Reply:
Looks like this: https://github.com/infracost/infracost/pull/1964/commits/3e140695e317aed174bbd16af4c6249b7f9feb88